### PR TITLE
Add notes for configuring Synology Diskstation

### PR DESCRIPTION
**Pull request by amil109**
I have a working set up for SSL enabled remote access on a Synology diskstation, taking advantage of the GUI as much as possible, to ensure automatic renewal of certs from Let's Encrypt, etc. It took me about 8 hours to suss it all out, but it could be achieved in about 30 mins if you knew exactly what to do... may not be widely useful, but since Synology is officially supported, I figured this might be a good addition.

There's also a minor error in the 'allow' masks - these should be 192.168.0.0/24 to allow access to anything in the 192.168.0.xxx range.

### CODE REVIEW

**Comment by Confectrician** [comment]
Hi @amil109 

Sorry for the delay, i am quite busy last days.
It will take some time for me, but looks really promising on a first quick look.

Could you do me a favor already and add a sign-off statement to your first post her in this pull request?

Info about [sign-off](https://github.com/openhab/openhab-docs/blob/master/CONTRIBUTING.md#sign-your-work) in the corresponding article.

Usually this has to be done by every commit, but for a first time contributor it is perfectly fine as comment in the pull request.
I will add it to the merge commit then manually.

Feel free to ask if there are any questions.

More Review feedback, when i had a closer look.

BR
Jerome

**Comment by amil109** [comment]
Hi Jerome,

No worries at all - added the comment! Let me know if you have any
questions about the updates.

Kind regards,

Andrew Mills

File: installation/security.md [hunk 1]
```diff
@@ -430,6 +431,149 @@ server {
     }
 }
 ```
+{: #synology-remote-config}
+#### Configuration on Synology DiskStation
+
+Synology DSM (as of 6.2) has the ability to automatically acquire certificates from Let's Encrypt and renew them every 90 days as required. The majority of the configuration mentioned above can be completed through the DSM GUI, but SSH access is required to implement authentication (**authentication is essential for remote access to your openHAB instance**).
```
Comment (Confectrician) [inline installation/security.md:437]: Please use one line per sentence.
The System will take care of a good presentation on the website.

File: installation/security.md [hunk 2]
```diff
@@ -430,6 +431,149 @@ server {
...
+Before you continue, make sure you have the below conditions:
+
+- A working installation of openHAB on your DiskStation (see the [Synology Installation Guide](https://www.openhab.org/docs/installation/synology.html/))
+- Your own domain you can configure the CAA record for (see [Setting up a Domain](#nginx-domain))
+- Access to your DiskStation by SSH ([How to login to DSM with root permission via SSH/Telnet](https://www.synology.com/en-global/knowledgebase/DSM/tutorial/General_Setup/How_to_login_to_DSM_with_root_permission_via_SSH_Telnet/))
+- Ports 80 and 443 forwarded from your router to your DiskStation (make sure you reconfigure the router web UI to a different port first, so you don't lose access!)
+
+Log into the GUI of your DiskStation as administrator, and open the package center. Install Apache HTTP Server. This is needed to generate the password files.
+
+Go to Control Panel > Application Portal > Reverse Proxy. We will set up two reverse proxies, one for HTTP and one for HTTPS. The HTTP one can be disabled later if desired (not at all essential if you will only use the app remotely, and never a browser).
```
Comment (Confectrician) [inline installation/security.md:448]: Please use one line per sentence.

File: installation/security.md [hunk 3]
```diff
@@ -430,6 +431,149 @@ server {
...
+|Destination Hostname:      |localhost        |
+|Destination Port:          |8080 (or whichever HTTP port your openHAB instance is on)|
+
+Verify that the reverse proxy is working as expected - try http://your-hostname.com and https://your-hostname.com - you should end up at the openHAB landing page in both cases, but will get a security warning for the https site.
+
+Next, acquire certificates from Let's Encrypt using the GUI in DSM.
+
+Go to Control Panel > Security > Certificate, and click on 'Add'.
+Select the option to 'Add a new Certificate'.
+Put in a description, something like 'openHAB SSL Cert' (it doesn't matter). Select 'Get a certificate from Let's Encrypt' and check the box to set it as default. Click next.
```
Comment (Confectrician) [inline installation/security.md:484]: Please use one line per sentence.

File: installation/security.md [hunk 4]
```diff
@@ -430,6 +431,149 @@ server {
...
+Put in your domain name and email address. Add a 'Subject Alternative Name' if you want a different topic in the subject line when Let's Encrypt email you about that certificate (not essential).
```
Comment (Confectrician) [inline installation/security.md:485]: Please use one line per sentence.

File: installation/security.md [hunk 5]
```diff
@@ -430,6 +431,149 @@ server {
...
+Click Apply, and wait a few minutes - your certificate is done!
+
+::: tip Note
+Sometimes you may receive an error at the end of the certificate wizard - the first time this happens, click on 'cancel and see if you have a certificate anyway. If the certifcate has been generated, you are good to go.
```
Comment (Confectrician) [inline installation/security.md:489]: Please use one line per sentence.

File: installation/security.md [hunk 6]
```diff
@@ -430,6 +431,149 @@ server {
...
+:::
+
+Select the certificate that has just been created, and click on 'Configure'.
+Ensure that the new certificate is listed next to your-hostname.com in the table - something like the below. If it's not selected, update it.
```
Comment (Confectrician) [inline installation/security.md:493]: Please use one line per sentence.

File: installation/security.md [hunk 7]
```diff
@@ -430,6 +431,149 @@ server {
...
+|:------------------------- |:--------------- |
+|your-hostname.com          |your-hostname.com|
+|FTPS                       |synology.com     |
+|Cloud Station Server       |synology.com     |
+|etc etc                    |synology.com     |
+
+Once this is done, update the CAA record for your-hostname.com with your registrar (exact process will vary by registrar).
+Within an hour or so, you should not receive the security warning for https://your-hostname.com.
+
+Next, you must add authentication to the reverse proxy. There's no GUI way to do this, so we need to create another small NGINX virtual host on the DiskStation.
```
Comment (Confectrician) [inline installation/security.md:505]: Please use one line per sentence.

File: installation/security.md [hunk 8]
```diff
@@ -430,6 +431,149 @@ server {
...
+
+Log into your DiskStation by SSH. Use the admin username and password.
```
Comment (Confectrician) [inline installation/security.md:507]: Please use one line per sentence.

File: installation/security.md [hunk 9]
```diff
@@ -430,6 +431,149 @@ server {
...
+Create a .htpasswd file in your openHAB userdata folder (your userdata location may vary, update accordingly):
+```shell
+htpasswd -c /volume1/SmartHome/openHAB/userdata/.htpasswd username
+```
+Next, add a very simple NGINX configuration similar to that created above, but without the SSL parameters. DSM comes with vi installed by default, but you may wish to [install nano](https://anto.online/other/how-to-install-nano-on-your-synology-nas/)
```
Comment (Confectrician) [inline installation/security.md:512]: Please use one line per sentence.

File: installation/security.md [hunk 10]
```diff
@@ -430,6 +431,149 @@ server {
...
+    }
+
+}
+```
+Once you are done, save the file, restart and test NGINX:
+
+```shell
+sudo nginx -s reload && sudo nginx -t
+```
+As above, the first part of the file redirects any HTTP queries to HTTPS directly. If you don't get any errors, update the reverse proxy settings in the DSM GUI to point to these new endpoints. Back in the GUI, go to Control Panel > Application Portal > Reverse Proxy, make the updates below:
```
Comment (Confectrician) [inline installation/security.md:556]: Please use one line per sentence.

**Pull request merged**